lib/improver/tests/nbhood/use_nbhood/test_ApplyNeighbourhoodProcessingWithAMask.py

```diff
@@ -161,6 +167,72 @@ def test_basic(self):
         self.assertEqual(result.data.shape, expected_shape)
         self.assertArrayAlmostEqual(result.data, expected)
 
+    def test_preserve_dimsensions_input(self):
+        """Test that the dimsensions on the output cube are the same as the
+           input cube, apart from the additional topographic zone coordinate.
+        """
+        self.cube.remove_coord("realization")
+        cubes = iris.cube.CubeList()
+        for i in range(3):
+            threshold_coord = DimCoord([i], long_name="threshold")
+            threshold_cube = iris.util.new_axis(self.cube.copy())
+            threshold_cube.add_dim_coord(threshold_coord, 0)
+            cubes.append(threshold_cube)
+        cube = cubes.concatenate_cube()
+        cubes = iris.cube.CubeList()
+        for i in range(4):
+            realization_coord = DimCoord([i], long_name="realization")
+            realization_cube = iris.util.new_axis(cube.copy())
+            realization_cube.add_dim_coord(realization_coord, 0)
+            cubes.append(realization_cube)
+        cube = cubes.concatenate_cube()
```

Comment:
Could/should the stuff above here either:
- be inserted into setUp?
- be made by an additional function so that one merely calls
```python
cube_with_extra_dims({'theshold': 3, 'realization': 4})

    def big_cube(self, new_dims):
        self.cube.remove_coord("realization")
        for dim_name, dim_size in new_dims.iteritems():
            cubes = iris.cube.CubeList()
            for i in range(dim_size):
                threshold_coord = DimCoord([i], long_name=dim_name)
                threshold_cube = iris.util.new_axis(self.cube.copy())
                threshold_cube.add_dim_coord(threshold_coord, 0)
                cubes.append(threshold_cube)
            cube = cubes.concatenate_cube()
        return cube
```